marathon/resource_marathon_app.go

```diff
@@ -375,7 +377,59 @@ func resourceMarathonApp() *schema.Resource {
 	}
 }
 
+type deploymentEvent struct {
+	id    string
+	state string
+}
+
+func readDeploymentEvents(meta interface{}, c chan deploymentEvent) error {
+	config := meta.(config)
+	client := config.Client
+
+	EventIDs := marathon.EventIDDeploymentSuccess | marathon.EventIDDeploymentFailed
+
+	events, err := client.AddEventsListener(EventIDs)
+	if err != nil {
+		log.Fatalf("Failed to register for events, %s", err)
+	}
+	defer client.RemoveEventsListener(events)
+
+	for {
+		select {
+		case event := <-events:
+			switch mEvent := event.Event.(type) {
+			case *marathon.EventDeploymentSuccess:
+				c <- deploymentEvent{mEvent.ID, event.Name}
+				return nil
+			case *marathon.EventDeploymentFailed:
+				c <- deploymentEvent{mEvent.ID, event.Name}
+				return errors.New("Received deployment_failed event from marathon")
```

Comment:
Since this is only ever finding one deployment event, would it be possible that there are multiple deployments happening and this ends on finding a deployment event that does not have the correct deployment id that we were waiting for?